attachment_synchronize_pattimpex/models/attachment_queue.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Akretion (http://www.akretion.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, fields, models
+
+
+class AttachmentQueue(models.Model):
+    _inherit = "attachment.queue"
+
+    export_id = fields.Many2one("ir.exports")
+    file_type = fields.Selection(
+        selection_add=[("import_pattern", "Import using Patterns")]
+    )
+    pattimpex_id = fields.Many2one(
+        "patterned.import.export",
+        readonly=True,
+        string="Patterned Import/Export",
+        help="The Patterned Import/Export (and its import job) that was created",
+    )
+
+    # Import part
+
+    def _run_import_pattern(self):
+        patterned_import = self.env["patterned.import.export"].create(
```

Comment:
TODO create a method on the patterned.import.export named "def enqueue" so you can share the code with the wizard